### ISSUE TITLE
Examining when ice formation and breakup occurs

### ISSUE THREAD

**Issue by dzrobert** [USGS-R/glmtools#251]
I used to use a command:
iceonoff_mod = mda.lakes::get_ice_onoff(ice_data, get_temp(nc_file))
to get ice information

This now gives me the following error:
Error in ice_runs[, "values"] : subscript out of bounds

Anyway to fix this so that I can compare actual and observed freeze and breakup dates?

**Comment by hdugan** [comment]
I'm not familiar with the mda.lakes package, but does 
`get_temp(nc_file)` still work on its own?

**Comment by jread-usgs** [comment]
Oh, I didn't realize this was an `mda.lakes` function. That package is pretty unsupported for general use...

The function is [here](https://github.com/USGS-R/mda.lakes/blob/afb436e047d2a9ca30dfdeae13745d2ee5109455/R/get_ice_onoff.R#L17). I'm guessing there is either a change w/ GLM v3 that isn't supported or part of the move to the updated `get_var`?

**Comment by dzrobert** [comment]
Yes, that works fine.  I was using the following series of commands to compare actual and measured ice freeze and breakup:
iceonoff_mod = mda.lakes::get_ice_onoff(ice_data, get_temp(nc_file))
#MODELED Ice Dates
iceonoff_mod
iceonoff_obs = read.csv('Green_ice19.csv', as.is=TRUE)
iceonoff_obs$on = as.POSIXct(iceonoff_obs$on)
iceonoff_obs$off = as.POSIXct(iceonoff_obs$off)
sprintf('Ice ON: %g mean diff mod - obs', mean(as.numeric(iceboth$on_mod - iceboth$on_obs, units='days')))
sprintf('Ice OFF: %g mean diff mod - obs', mean(as.numeric(iceboth$off_mod - iceboth$off_obs, units='days')))

sprintf('Ice ON: %g mean absolute diff', mean(abs(as.numeric(iceboth$on_mod - iceboth$on_obs, units='days'))))
sprintf('Ice OFF: %g mean absolute diff', mean(abs(as.numeric(iceboth$off_mod - iceboth$off_obs, units='days'))))

**Comment by hdugan** [comment]
The error is in the `mda.lakes::get_ice_onoff` function. 
It was probably due to an update in the `rle2` function. 
The function is calling the wrong names from the rle2 output. They need to be 
'value' 
'start' 
'stop' 
'length'
NOT values starts stops lengths

**Comment by hdugan** [comment]
This is a nice function though - maybe it should be in glmtools?

**Comment by dzrobert** [comment]
For us ice people it would be great.
Dale

**Comment by jread-usgs** [comment]
I'm worried about exposing it in the package because it can give you the wrong answer (it makes assumptions about ice dates relative to peak temperatures). I think it is right 99.9% of the time, but consider it to be more of an analysis method that you'd do on the data coming out of `get_ice()` (or _var) where you'd have control and visibility in the decision of when to declare on vs off. 

Or perhaps we expose it but treat as experimental?

**Comment by pchanson** [comment]
If it’s included, could it be called an ‘estimator’ or some other similar term that indicates it’s derived from assumed relationships that really are statistical in nature?  Alternatively, is there a need for more of a GLM data analysis package, kind of lake the existing lake physics package you created, Jordan?
---------------------------------------------
Paul Hanson
Distinguished Research Professor
UW Center for Limnology

**Comment by dzrobert** [comment]
I would be OK working with an experimental package?  Right now I am just trying to see if ice on and off dates are close, Rather than looking through the daily ice thicknesses, this was working great.  Is there a way just to grab a few lines from mda.lakes that I can run?

**Comment by hdugan** [comment]
Dale, I forked the mda repo and fixed the function. You can use it here:

`install_github('hdugan/mda.lakes')`

Don't know if you want a pull request to the main repo
